test/cpp/end2end/grpclb_end2end_test.cc

```diff
@@ -191,6 +192,13 @@ class BalancerServiceImpl : public BalancerService {
         shutdown_(false) {}
 
   Status BalanceLoad(ServerContext* context, Stream* stream) override {
+    for (const auto& entry : context->client_metadata()) {
```

Comment:
I've verified that the balancer would receive call creds metadata if `grpclb_channel_secure.cc` didn't call `grpc_channel_credentials_duplicate_without_call_credentials`. This is because the [server_auth_filter only removes metadata if the channel creds provide a processor](https://github.com/grpc/grpc/blob/master/src/core/lib/security/transport/server_auth_filter.cc#L169), but the fake credentials don't.

I spent quite a bit of time trying to come up with a way to have a negative test, but short of having a mock for `grpclb_channel_secure.cc` that would not remove the call creds, I don't know how to put together a negative test.